### PR TITLE
Add B Functions for Testing PID Settings

### PR DESCRIPTION
**Pull request by krkeegan**
Adds B13&14 functions which allow a user to request a series of stepped movements on a particular axis.  During these movements the error of the axis during the movement is reported on a series of lines which can be graphed by the user.

These functions are somewhat rudimentary, but they are very helpful for initial testing of PID settings.  

I have started to write some documentation of the Tuning process here:

https://github.com/MaslowCNC/Firmware/wiki/PID-Tuning

### CODE REVIEW

**Comment by krkeegan** [comment]
I have horded too many commits while working on my PID testing.  I think it would be helpful to pull out smaller chunks for review and testing, particularly since we seem to be having issues with the larger collection.

**Comment by BarbourSmith** [comment]
That's a great idea. I hard coded something similar when I was testing then commented it out. Your method seems better 😁

File: cnc_ctrl_v1/CNC_Functions.h [hunk 1]
```diff
@@ -1192,6 +1192,111 @@ void  setSpindlePower(boolean powerState) {
...
+        if (i > 0){
+            speed = start + (span * (i/(steps-1)));
+        }
+        startTime = micros();
+        axis->motorGearboxEncoder.write(speed);
+        while (startTime + 2000000 > current){
+          if (current - print > 20000){
+            // Calculate and log error on same frequency as PID interrupt
+            distMoved   =  axis->motorGearboxEncoder.encoder.read() - lastPosition;
+            reportedSpeed= (7364.0*distMoved)/float(current - print);  //6*10^7 us per minute, 8148 steps per revolution
```
Comment (BarbourSmith) [inline cnc_ctrl_v1/CNC_Functions.h]: This 7364 was replaced with a calculated value recently that might be better to use, not a big issue

**Comment by BarbourSmith** [comment]
This one passes all my tests and seems ready to merge. Anyone else have any issues?

**Pull request merged**